### ISSUE TITLE
ErrorException occurred only in v0.8.12

### ISSUE THREAD

**Issue by yappkahowe** [bobthecow/psysh#430]
I always use psysh with laravel/tinker. Today after I updated psysh from 0.8.11 to 0.8.12, I got the error below when I run `php artisan tinker`.

` rmdir(C:\Users\<User>\AppData\Local\Temp\php-xdg-runtime-dir-fallback-): Directory not empty`

The error disappears after I downgraded it back to v0.8.11. If I manually delete the `php-xdg-runtime-dir-fallback-`, I was able to run `php artisan tinker` once. Then I run `php artisan tinker` again, the error comes back.

**Comment by bobthecow** [comment]
Thanks for reporting this. It looks like a Windows incompatibility in an underlying library. I'll work on a fix, but in the meantime, you should be able to work around it by creating a directory (I'd use something like `C:\Users\<User>\AppData\Local\Temp\psysh-runtime-dir-fallback` or something) and setting the `XDG_RUNTIME_DIR` environment variable to that path.

**Comment by bobthecow** [comment]
(reopening because this isn't actually solved yet)

**Comment by abdolrhman** [comment]
am having the same problem , am using laravel55 
every thing was working fine , even  tinker ,
when i used php artisan make:transformer 
the Error come to life !!! 
  "rmdir(C:\Users\<user>\AppData\Local\Temp\php-xdg-runtime-dir-fallback-): Directory not empty"

**Comment by bobthecow** [comment]
`composer update` should fix it. This was fixed in v0.8.13.

**Comment by yappkahowe** [comment]
Tested the fix in v0.8.13. It worked. Thank you very much for the fix.

**Comment by bobthecow** [comment]
Can any of you tell me what flavor of PHP you have installed on Windows? Everything works as expected when installing PHP via XAMPP and WSL. I haven't been able to reproduce this issue. The following would be super helpful, if any/all of you can provide it:

- Windows version
- App you're using for a shell (`cmd.exe`, etc)
- PHP version, and how it was installed
- The output of `psy\info()`

Thanks!

**Comment by yappkahowe** [comment]
* Windows 10 Pro
* Powershell
* PHP 7.17, XAMPP 7.1.7

##### Output of `psy\info()`

[
     "PsySH version" => "v0.8.13",
     "PHP version" => "7.1.7",
     "default includes" => [],
     "require semicolons" => false,
     "error logging level" => 32767,
     "config file" => [
       "default config file" => null,
       "local config file" => null,
       "PSYSH_CONFIG env" => false,
     ],
     "updates" => [
       "update available" => true,
       "latest release version" => "v0.8.13",
       "update check interval" => "never",
       "update cache file" => "C:/Users/<User>/AppData/Roaming/PsySH/update_check.json",
     ],
     "pcntl" => [
       "pcntl available" => false,
       "posix available" => false,
     ],
     "readline" => [
       "readline available" => true,
       "readline enabled" => true,
       "readline service" => "Psy\Readline\Libedit",
     ],
     "history" => [
       "history file" => "C:/Users/<User>/AppData/Roaming/PsySH/psysh_history",
       "history size" => null,
       "erase duplicates" => null,
     ],
     "docs" => [
       "manual db file" => null,
       "sqlite available" => true,
     ],
     "autocomplete" => [
       "tab completion enabled" => true,
       "custom matchers" => [],
       "bracketed paste" => false,
     ],
   ]

**Comment by bobthecow** [comment]
Thanks, I was able to reproduce it. I think I have a fix!

**Comment by bobthecow** [comment]
Yep, looks good on my Windows VM. If any of you can test the `develop` branch to verify, that'd be awesome. The fix will go out in the next stable release!

**Comment by abdolrhman** [comment]
Good Job bob , and good luck

**Comment by potsed** [comment]
I know this issue is closed, but it is occurring again on 0.8.14  PHP 7.1.7

![image](https://user-images.githubusercontent.com/542347/32532593-1bb90ac2-c497-11e7-8458-707e4721a054.png)

Output of psy\info()

```php
[
     "PsySH version" => "v0.8.14",
     "PHP version" => "7.1.7",
     "default includes" => [],
     "require semicolons" => false,
     "error logging level" => 32767,
     "config file" => [
       "default config file" => null,
       "local config file" => null,
       "PSYSH_CONFIG env" => false,
     ],
     "updates" => [
       "update available" => false,
       "latest release version" => "v0.8.14",
       "update check interval" => "never",
       "update cache file" => "~/.config/psysh/update_check.json",
     ],
     "pcntl" => [
       "pcntl available" => true,
       "posix available" => true,
     ],
     "readline" => [
       "readline available" => true,
       "readline enabled" => true,
       "readline service" => "Psy\Readline\Libedit",
       "readline library" => "EditLine wrapper",
     ],
     "history" => [
       "history file" => "~/.config/psysh/psysh_history",
       "history size" => null,
       "erase duplicates" => null,
     ],
     "docs" => [
       "manual db file" => null,
       "sqlite available" => true,
     ],
     "autocomplete" => [
       "tab completion enabled" => true,
       "custom matchers" => [],
       "bracketed paste" => false,
     ],
   ]
```

**Comment by bobthecow** [comment]
Yours doesn't look like quite the same issue, but probably related.

Can you paste the output from `lstat("/tmp/php-xdg-runtime-dir-fallback-ec2-user")`?

**Comment by deeem** [comment]
I think i got the same error.

Docker container with Debian 8.

```
root@de49f09d8bf5:/var/www# php artisan tinker                                                
  [ErrorException]
  rmdir(/tmp/php-xdg-runtime-dir-fallback-): Directory not empty 
```
psy/info()
```php
[
     "PsySH version" => "v0.8.14",
     "PHP version" => "7.1.11",
     "default includes" => [],
     "require semicolons" => false,
     "error logging level" => 32767,
     "config file" => [
       "default config file" => null,
       "local config file" => null,
       "PSYSH_CONFIG env" => false,
     ],
     "updates" => [
       "update available" => false,
       "latest release version" => "v0.8.14",
       "update check interval" => "weekly",
       "update cache file" => "~/.config/psysh/update_check.json",
     ],
     "pcntl" => [
       "pcntl available" => false,
       "posix available" => true,
     ],
     "readline" => [
       "readline available" => true,
       "readline enabled" => true,
       "readline service" => "Psy\Readline\Libedit",
       "readline library" => "EditLine wrapper",
     ],
     "history" => [
       "history file" => "~/.config/psysh/psysh_history",
       "history size" => null,
       "erase duplicates" => null,
     ],
     "docs" => [
       "manual db file" => null,
       "sqlite available" => true,
     ],
     "autocomplete" => [
       "tab completion enabled" => true,
       "custom matchers" => [],
       "bracketed paste" => false,
     ],
   ]
```
 lstat('/tmp/php-xdg-runtime-dir-fallback-');
```php
[
     0 => 61,
     1 => 12059210,
     2 => 16832,
     3 => 3,
     4 => 0,
     5 => 0,
     6 => 0,
     7 => 4096,
     8 => 1510162329,
     9 => 1510162329,
     10 => 1510162329,
     11 => 4096,
     12 => 8,
     "dev" => 61,
     "ino" => 12059210,
     "mode" => 16832,
     "nlink" => 3,
     "uid" => 0,
     "gid" => 0,
     "rdev" => 0,
     "size" => 4096,
     "atime" => 1510162329,
     "mtime" => 1510162329,
     "ctime" => 1510162329,
     "blksize" => 4096,
     "blocks" => 8,
   ]
```

**Comment by potsed** [comment]
Here's my lstat output @bobthecow 

```
>>> lstat("/tmp/php-xdg-runtime-dir-fallback-ec2-user")
=> [
     0 => 51713,
     1 => 275891,
     2 => 16832,
     3 => 3,
     4 => 500,
     5 => 500,
     6 => 0,
     7 => 4096,
     8 => 1510182428,
     9 => 1510182428,
     10 => 1510182428,
     11 => 4096,
     12 => 8,
     "dev" => 51713,
     "ino" => 275891,
     "mode" => 16832,
     "nlink" => 3,
     "uid" => 500,
     "gid" => 500,
     "rdev" => 0,
     "size" => 4096,
     "atime" => 1510182428,
     "mtime" => 1510182428,
     "ctime" => 1510182428,
     "blksize" => 4096,
     "blocks" => 8,
   ]
```

**Comment by renatoagomes** [comment]
Temporary fix, thanks to @thiagobraga:

In my case, [laravel/tinker](github.com/laravel/tinker) uses this package and theirs composer.json looks like this:

![image](https://user-images.githubusercontent.com/9811888/32758589-3e8e4fe2-c8cc-11e7-9ac9-6390a8f8362b.png)

So in order to fix this issue, we can change our own composer.json to get the specific version v0.8.11

      "laravel/tinker": "~1.0",
      "psy/psysh": "0.8.11",

worked for me :)

**Comment by bobthecow** [comment]
@deeem @potsed what does `getmyuid()` say for both of you on these machines?

**Comment by deeem** [comment]
```php
>>> getmyuid();
=> 1000
```

**Comment by bobthecow** [comment]
@deeem for some reason your psysh temp directory is owned by root, and you clearly aren't root.

**Comment by bobthecow** [comment]
Okay. I've pushed a really aggressive workaround to the `develop` branch, and plan on cutting a stable release ASAP. If you can test that branch before I do, that'd be great!

**Comment by deeem** [comment]
Yes. It fix issue in my case.

**Comment by svenluijten** [comment]
I can also confirm that this is fixed since a45bad00a418f396a2aa900e591eba6b21dc6398! For anyone confused / want to try it: `composer require psy/psych:dev-develop`

**Comment by bobthecow** [comment]
No need for `dev-develop`! This fix was [released in v0.8.15](https://github.com/bobthecow/psysh/releases/tag/v0.8.15) :)
